x/ecocredit/migrations/v3/patch.go

```diff
@@ -0,0 +1,190 @@
+package v3
+
+import (
+	"context"
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	basketapi "github.com/regen-network/regen-ledger/api/regen/ecocredit/basket/v1"
+	api "github.com/regen-network/regen-ledger/api/regen/ecocredit/v1"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+// patchMigrate performs following migrations
+// - add reference id to existing projects
+// - update issuance date for credit batches
+// - update curator address for baskets
+func patchMigrate(ctx context.Context, sdkCtx sdk.Context, ss api.StateStore,
+	basketStore basketapi.StateStore, oldBatchDenomToNewDenomMap map[string]string) error {
+	if sdkCtx.ChainID() == "regen-1" {
+		return patchMainnet(ctx, ss, oldBatchDenomToNewDenomMap)
+	} else if sdkCtx.ChainID() == "regen-redwood-1" {
+		return patchRedwood(ctx, ss, basketStore, oldBatchDenomToNewDenomMap)
+	}
+
+	return nil
+}
+
+func patchMainnet(ctx context.Context, ss api.StateStore, oldBatchDenomToNewDenomMap map[string]string) error {
+	// project location -> reference-id
+	// CD-MN -> "" // TODO: add reference-id
+	// KE    -> ""
+
+	locationToReferenceIdMap := make(map[string]string)
+	locationToReferenceIdMap["CD-MN"] = ""
+	locationToReferenceIdMap["KE"] = ""
+
+	// add reference id to existing projects
+	if err := addReferenceIds(ctx, ss, locationToReferenceIdMap); err != nil {
+		return err
+	}
+
+	// batch issuance dates
+	//  C01-20190101-20191231-001  -  "2022-05-06T01:33:13Z"
+	//  C01-20190101-20191231-002  -  "2022-05-06T01:33:19Z"
+	//  C01-20190101-20191231-003  -  "2022-05-06T01:33:25Z"
+	//  C01-20190101-20191231-004  -  "2022-05-06T01:33:31Z"
+
+	batchIdToIssuanceDateMap := make(map[string]string)
+	batchIdToIssuanceDateMap["C01-20190101-20191231-001"] = "2022-05-06T01:33:13Z"
+	batchIdToIssuanceDateMap["C01-20190101-20191231-002"] = "2022-05-06T01:33:19Z"
+	batchIdToIssuanceDateMap["C01-20190101-20191231-003"] = "2022-05-06T01:33:25Z"
+	batchIdToIssuanceDateMap["C01-20190101-20191231-004"] = "2022-05-06T01:33:31Z"
+	// update issuance date for credit batches
+	if err := updateBatchIssueanceDate(ctx, ss, oldBatchDenomToNewDenomMap, batchIdToIssuanceDateMap); err != nil {
+		return err
+	}
+
+	// we don't have baskets on mainnet
+
+	return nil
+}
+
+func patchRedwood(ctx context.Context, ss api.StateStore,
+	basketStore basketapi.StateStore, oldBatchDenomToNewDenomMap map[string]string) error {
+	// project location -> reference-id
+	// FR              -> "" // TODO: add reference-id
+	// US              -> ""
+	// AU-NSW 2453     -> ""
+	// K               -> ""
+	// US-FL 9876      -> ""
```

Comment:
Same as above. Will check with @S4mmyb on reference id for projects created on redwood.

Reply:
My guess is that we can leave reference ID blank on redwood.

Reply:
Sam is checking with the interface team. They may want some of these projects to have a reference id but we can handle in a followup.

Reply:
We can move forward without including any sort of reference id for projects on redwood @ryanchristo